packages/sdk/src/apiClient/basicAuthProvider.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AxiosRequestConfig } from "axios";
+import { AuthProvider } from "./authProvider";
+import { ErrorWithCode, ErrorCode, ErrorMessage } from "../core/errors";
+import { formatString } from "../util/utils";
+
+/**
+ * Provider that handles Basic authentication
+ *
+ * @beta
+ */
+export class BasicAuthProvider implements AuthProvider {
+  private userName: string;
+  private password: string;
+
+  /**
+   *
+   * @param userName - Username used in basic auth
+   * @param password - Password used in basic auth
+   *
+   * @beta
+   */
+  constructor(userName: string, password: string) {
+    if (!userName) {
+      throw new ErrorWithCode(
+        formatString(ErrorMessage.ArgumentEmpty, "username"),
+        ErrorCode.InvalidParameter
+      );
+    }
+    if (!password) {
+      throw new ErrorWithCode(
+        formatString(ErrorMessage.ArgumentEmpty, "password"),
```

Comment:
since the error code is `InvalidParameter`, should we use `parameter` instead of `argument` in the error message?

Reply:
Fixed.